waterlog/src/Components/Map.js

```diff
@@ -1,148 +1,95 @@
+  let monitorMapData = monitorsCoordinates.map(mon => [mon.lat, mon.long, levelToIntensity(mon.faultLevel, maxIntensity)]);
+  let segmentMapData = segmentCoordinates.map(seg => [seg.lat, seg.long, levelToIntensity(seg.faultLevel, maxIntensity)]);
 
-
-function generateMapIcons({ segments, markers }, simpleView) {
-  const defaultColor = simpleView ? backgroundColor : lighterColor;
-  if (!segments || !markers) {
-    return <div />;
-  }
-  return segments.map(segment => {
-    const sensorIn = markers.find(marker => marker.id === segment.senseIDIn);
-    const sensorOut = markers.find(marker => marker.id === segment.senseIDOut);
-    if (!sensorIn || !sensorOut) {
-      return <div />;
-    }
-    const validSensor = ["id", "lat", "lon", "status"];
-
-    if (
-      Object.keys(sensorIn) === validSensor ||
-      Object.keys(sensorOut) === validSensor
-    ) {
-      return <div />;
-    }
-
-    let sensorInColor = defaultColor,
-      sensorOutColor = defaultColor,
-      segmentColor = defaultColor;
-    if (sensorIn.status.toLowerCase() === "faulty") {
-      sensorInColor = errorColor;
-    }
-    if (sensorOut.status.toLowerCase() === "faulty") {
-      sensorOutColor = errorColor;
-    }
-    if (segment.status.toLowerCase() === "leak") {
-      segmentColor = errorColor;
-    }
-
-    return (
-      <div>
-        <Polyline
-          positions={[
-            [sensorIn.lat, sensorIn.lon],
-            [sensorOut.lat, sensorOut.lon]
-          ]}
-          color={segmentColor}
-          weight={7}
-        >
-          <Popup>
-            <span>
-              {"segment " + segment.id + "\n status " + segment.status}
-            </span>
-          </Popup>
-        </Polyline>
-        <CircleMarker
-          center={[sensorIn.lat, sensorIn.lon]}
-          radius={cicrle_number}
-          opacity={0.7}
-          key={sensorIn.id}
-          color={sensorInColor}
-          fillOpacity={1}
-        >
-          <Popup>
-            <span>
-              {"sensor " + sensorIn.id + "\n status " + sensorIn.status}
-            </span>
-          </Popup>
-        </CircleMarker>
-        <CircleMarker
-          fill={true}
-          center={[sensorOut.lat, sensorOut.lon]}
-          radius={cicrle_number}
-          opacity={0.7}
-          key={sensorOut.id}
-          color={sensorOutColor}
-          fillOpacity={1}
-        >
-          <Popup>
-            <span>
-              {"sensor " + sensorOut.id + "\n status " + sensorOut.status}
-            </span>
-          </Popup>
-        </CircleMarker>
-      </div>
-    );
-  });
+  let heatMapData = monitorMapData.concat(segmentMapData);
+  return heatMapData;
 }
 class MapComponent extends Component {
   componentDidMount() {
     this.props.fetchMapsData();
+    this.props.fetchHeatMapsData();
   }
 
   constructor(props) {
     super(props);
     this.state = {
       lat: -25.783425,
-			lng: 28.336046,
+      lng: 28.336046,
```

Comment:
Will it not be easier to have these in constants?

Reply:
I am setting default values, do I still need to put them in variables

Reply:
I moved the values to a constant